ldk/node/src/hostClients/windowClient.test.ts

```diff
@@ -0,0 +1,171 @@
+import { mocked } from 'ts-jest/utils';
+import createMockInstance from 'jest-create-mock-instance';
+import * as Services from '../grpc/window_grpc_pb';
+import * as Messages from '../grpc/window_pb';
+import { ConnInfo } from '../grpc/broker_pb';
+import { WindowClient } from './windowClient';
+import { Session } from '../grpc/session_pb';
+import { Logger } from '../logging';
+import {
+  captureMockArgument,
+  createCallbackHandler,
+  createStreamingHandler,
+  createWaitHandler,
+  defaultConnInfo,
+  defaultSession,
+  identityCallback,
+} from '../jest.helpers';
+import { WindowInfoResponse } from './windowService';
+
+jest.mock('../grpc/window_grpc_pb');
+
+const MockClientClass = mocked(Services.WindowClient);
+const logger = new Logger('test-logger');
+
+describe('WindowClient', () => {
+  let subject: WindowClient;
+  let mockGRPCClient: jest.Mocked<Services.WindowClient>;
+  let connInfo: ConnInfo.AsObject;
+  let session: Session.AsObject;
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+    subject = new WindowClient();
+    connInfo = defaultConnInfo;
+    session = defaultSession;
+
+    mockGRPCClient = createMockInstance(Services.WindowClient);
+    mockGRPCClient.waitForReady.mockImplementation(createWaitHandler());
+    MockClientClass.mockImplementation(() => mockGRPCClient as any);
+
+    await expect(
+      subject.connect(connInfo, session, logger),
+    ).resolves.toBeUndefined();
+  });
+
+  describe('#queryActiveWindow', () => {
+    let sentRequest: Messages.WindowActiveWindowRequest;
+    let sentResponse: Messages.WindowActiveWindowResponse;
+    let queryResult: Promise<WindowInfoResponse>;
+
+    beforeEach(async () => {
+      sentResponse = new Messages.WindowActiveWindowResponse().setWindow(
+        new Messages.WindowInfo(),
+      );
+
+      mockGRPCClient.windowActiveWindow.mockImplementation(
+        createCallbackHandler(sentResponse),
+      );
+
+      queryResult = subject.queryActiveWindow();
+
+      sentRequest = captureMockArgument(mockGRPCClient.windowActiveWindow);
+    });
+
+    it('should return a transformed response', async () => {
+      await expect(queryResult).resolves.toMatchObject({
```

Comment:
I'd like to be able to build a helper for `toMatchInterface` as this will likely get brittle quickly, but I'm not sure the JSR (juice to squeeze ratio) is there at the moment.

Reply:
IMO sometimes brittleness isn't the worst thing in tests; it helps us identify when something changes.